src/components/HelloWorld.vue

```diff
@@ -1,57 +1,212 @@
+        <b-form-input class="input" type="password" placeholder="Enter your Password" v-model="user.password"></b-form-input>
+        <b-form-input class="input" type="password" placeholder="Enter your Confirm-Password" v-model="user.con_password"></b-form-input>
+        <b-form-checkbox id="checkbox1" v-model="status" value="accepted" unchecked-value="not_accepted">
+          I accept the terms and use
+        </b-form-checkbox>
+      </form>
+    </b-modal>
   </div>
 </template>
 
 <script>
-export default {
-  name: 'HelloWorld',
-  props: {
-    msg: String
-  }
-}
+  export default {
+    name: "Form",
+    data() {
+      return {
+        users: [],
+        userCount: 0,
+        status: false,
+        user: {
+          email: "",
+          name: "",
+          dob: "",
+          password: "",
+          con_password: ""
+        },
+        isEdit: false,
+        sortKey: "",
+        sortSettings: [{
+            email: true
+          },
+          {
+            name: true
+          },
+          {
+            dob: true
+          }
+        ],
+        desc: true
+      };
+    },
+    methods: {
+      clearName() {
+        this.name = "";
+      },
+      clearEmail() {
+        this.email = "";
+      },
+      clearDob() {
+        this.dob = "";
+      },
+      clearPassword() {
+        this.password = "";
+      },
+      clearConfirmPassword() {
+        this.con_password = "";
+      },
+      handleOk(evt) {
+        // Prevent modal from closing
+        evt.preventDefault();
+        if (!this.user.email) {
+          alert("Please enter your email");
+          return false;
+        }
+        if (!this.user.name) {
+          alert("Please enter your name");
+          return false;
+        }
+        if (!this.user.dob) {
+          alert("Please enter your DOB");
+          return false;
+        }
+        if (!this.user.password) {
+          alert("Please enter your Password");
+          return false;
+        }
+        if (!this.user.con_password) {
+          alert("Please enter your Confirm Password");
+          return false;
+        }
+        if (!this.status) {
+          alert("please accept our terms and conditions");
+          return false;
+        } else {
+          this.handleSubmit();
+          return true;
+        }
+      },
+      handleSubmit() {
+        if (this.isEdit) {
+          for (var i = 0; i < this.users.length; i++) {
+            if (this.users[i].id == this.user["id"]) {
+              this.users[i] = this.user;
+            }
+          }
+        } else {
+          this.users.push({
+            id: ++this.userCount,
+            email: this.user.email,
+            name: this.user.name,
+            dob: this.user.dob,
+            password: this.user.password,
+            con_password: this.user.con_password
+          });
+        }
+        this.$refs.modal.hide();
+        alert("form submitted successfully");
```

Comment:
don't use alert at all from now. alert is just to debug this. the final app should not have alert